### PR TITLE
[Docs] Expand custom plugin guide

### PR DESCRIPTION
**Pull request by dotNetkow**
Based on learnings from Cap 2.0 webinar, add more details about project setup/structure and notes on how to develop a plugin locally.

### CODE REVIEW

File: site/docs-md/plugins/index.md [hunk 1]
```diff
@@ -73,6 +77,85 @@ export interface EchoPlugin {
...
+}
+```
+
+Implement the web implementation in `src/web.ts`:
+
+```typescript
+async openMap(location: { latitude: number, longitude: number}): Promise<void> {
+  // logic here
+}
+```
```
[the remark this answers is not in the document]
Comment (mlynch) [reply site/docs-md/plugins/index.md:95]: I also think we need to address how to actually make the web side load in the app: importing the package.

Ex:

```typescript
import 'my-plugin';
```

File: site/docs-md/plugins/ios.md [hunk 2]
```diff
@@ -17,7 +17,7 @@ contributors:
 
 To get started, first generate a plugin as shown in the [Getting Started](./#getting-started) section of the Plugin guide.
 
-Next, open `your-plugin/ios/Plugin.xcworkspace` in Xcode.
```
[the remark this answers is not in the document]
Comment (dotNetkow) [reply site/docs-md/plugins/ios.md:20]: Hmm, when I open it via Xcode File -> Open, there's no workspace to open - it's a project (xcodeproj) hence why I changed it here.

File: site/docs-md/plugins/index.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (dotNetkow) [reply site/docs-md/plugins/index.md:95]: In this context, i'm showing how to add new functionality to a plugin, one new function. I purposely don't have all the other details in there, as they are a distraction.  But you make good points about the Web side of things, so I updated web.md (the PWA plugin page) with more detail such as registerWebPlugin.  Also, the plugin imports are there now, under Usage. I almost added it to the intro page but I think it's best in the Web-specific page on further thought.

File: site/docs-md/plugins/ios.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (hemangsk) [reply site/docs-md/plugins/ios.md:20]: @dotNetkow  There definitely is a Plugin.xcworkspace file inside ```your-plugin/ios/``` along with an xcodeproj file, and _both_ of them show up on Xcode File -> Open, therefore the original sentence seems totally correct.

(sorry for jumping out of nowhere, super excited for this MR)

**Comment by dotNetkow** [comment]
Moving over to new Capacitor site/docs repo to a [new PR here](https://github.com/ionic-team/capacitor-site/pull/5).

**Pull request closed without merging**